# Projekt DS & KI
Gruppenmitglieder: Clemens Fleischhauer, Nils Groetelaers, Nikita Helwer, Carl Placek, Julian Krupp

# Unser Datensatz:

https://archive.ics.uci.edu/dataset/186/wine+quality

# SMART-Ziel:

Wir wollen untersuchen, welchen Einfluss der Alkohol-Prozentgehalt auf die Weinqualität hat. Der UCI-Weindatensatz mit ca. 5000 Weißwein- und ca. 1500 Rotweinproben liefert ausreichend Daten, um erste Schlüsse zu ziehen. Unser Ziel ist die Analyse von Korrelationen und weiteren Zusammenhängen, sowie deren Schilderung und Dokumentation innerhalb der vorgegebenen Zeit (bis 23.11.).

# Quellen:

Titel: Modellierung der Weinqualität anhand physiochemischer Eigenschaften

https://cs229.stanford.edu/proj2019aut/data/assignment_308832_raw/25895690.pdf

Erkenntnisse aus der Quelle: Anhand der vier Größen chlorides, total sulfur dioxide, fixed acidity und pH oder chlorides, total sulfur dioxide, volatile acidity und free sulfur dioxide lässt sich die Weinqualität mithilfe eines Neural Network Models mit einer 99%-igen Sicherheit voraussagen. 

Wir sehen also: Alkohol scheint sich im Vergleich zu anderen Parametern nicht so gut zu eignen, um die Weinqualität vorauszusagen. Daher hat Alkohol vermutlich einen weniger starken Einfluss auf die Weinqualität als die anderen zur Vorhersage benutzten Parameter.

# Aufgabe 1:

In [1]:
# Datensatz Weißwein
import pandas as pd
ww = pd.read_csv('winequality-white.csv', sep=';')

print(ww.head())

print("=====================================================================================================================================>")

print(ww.columns) # 12 Spalten

print("=====================================================================================================================================>")

print(ww.index)   # 4898 Zeilen

print("=====================================================================================================================================>")

ww.describe()

   fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6 

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    4898.000000       4898.000000  4898.000000     4898.000000   
mean        6.854788          0.278241     0.334192        6.391415   
std         0.843868          0.100795     0.121020        5.072058   
min         3.800000          0.080000     0.000000        0.600000   
25%         6.300000          0.210000     0.270000        1.700000   
50%         6.800000          0.260000     0.320000        5.200000   
75%         7.300000          0.320000     0.390000        9.900000   
max        14.200000          1.100000     1.660000       65.800000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  4898.000000          4898.000000           4898.000000  4898.000000   
mean      0.045772            35.308085            138.360657     0.994027   
std       0.021848            17.007137             42.498065     0.002991   
min       0.009000             2.000000              9.000000     0.987110   
25%       0.036000            23.000000            108.000000     0.991723   
50%       0.043000            34.000000            134.000000     0.993740   
75%       0.050000            46.000000            167.000000     0.996100   
max       0.346000           289.000000            440.000000     1.038980   

                pH    sulphates      alcohol      quality  
count  4898.000000  4898.000000  4898.000000  4898.000000  
mean      3.188267     0.489847    10.514267     5.877909  
std       0.151001     0.114126     1.230621     0.885639  
min       2.720000     0.220000     8.000000     3.000000  
25%       3.090000     0.410000     9.500000     5.000000  
50%       3.180000     0.470000    10.400000     6.000000  
75%       3.280000     0.550000    11.400000     6.000000  
max       3.820000     1.080000    14.200000     9.000000

In [2]:
# Datensatz Rotwein
import pandas as pd
rw = pd.read_csv('winequality-red.csv', sep=';')

print(rw.head())

print("=====================================================================================================================================>")

print(rw.columns) # 12 Spalten

print("=====================================================================================================================================>")

print(rw.index)   # 1599 Zeilen

print("=====================================================================================================================================>")

rw.describe()

   fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5 

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    1599.000000       1599.000000  1599.000000     1599.000000   
mean        8.319637          0.527821     0.270976        2.538806   
std         1.741096          0.179060     0.194801        1.409928   
min         4.600000          0.120000     0.000000        0.900000   
25%         7.100000          0.390000     0.090000        1.900000   
50%         7.900000          0.520000     0.260000        2.200000   
75%         9.200000          0.640000     0.420000        2.600000   
max        15.900000          1.580000     1.000000       15.500000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  1599.000000          1599.000000           1599.000000  1599.000000   
mean      0.087467            15.874922             46.467792     0.996747   
std       0.047065            10.460157             32.895324     0.001887   
min       0.012000             1.000000              6.000000     0.990070   
25%       0.070000             7.000000             22.000000     0.995600   
50%       0.079000            14.000000             38.000000     0.996750   
75%       0.090000            21.000000             62.000000     0.997835   
max       0.611000            72.000000            289.000000     1.003690   

                pH    sulphates      alcohol      quality  
count  1599.000000  1599.000000  1599.000000  1599.000000  
mean      3.311113     0.658149    10.422983     5.636023  
std       0.154386     0.169507     1.065668     0.807569  
min       2.740000     0.330000     8.400000     3.000000  
25%       3.210000     0.550000     9.500000     5.000000  
50%       3.310000     0.620000    10.200000     6.000000  
75%       3.400000     0.730000    11.100000     6.000000  
max       4.010000     2.000000    14.900000     8.000000

## Anzahl Zeilen und Spalten und deren Bedeutung

### Datensatz Rotwein

Anzahl Zeilen: 1599

Anzahl Spalten: 12

### Datensatz Weißwein

Anzahl Zeilen: 4898 

Anzahl Spalten: 12

### Spalten (gleich für beide Datensätze):

"fixed acidity" = g/L, stabile Säuren, bestimmen den pH-Wert und Geschmack

"volatile acidity" = g/L, in geringen Mengen für Aroma gewünscht, zu viel ist ein Weinfehler

"citric acid" =  g/L, Teil der fixen Säuren, trägt zur Frische bei

"residual sugar" = g/L, unvergorener Zucker im Wein

"chlorides" = g/L, Salze, die den Salzgehalt beeinflussen

"free sulfur dioxide" = mg/L, schützt Wein vor Oxidation und unerwünschten Mikroorganismen

"total sulfur dioxide" = mg/L, Summe aus freiem und gebundenem SO2, Gesamtschutz des Weins

"density" = g/cm3, Dichte des Weins

"pH" = von 1 bis 14, 1 - 7 heißt sauer, 7 - 14 ist basisch

"sulphates" = g/L, Salze der Schwefelsäure, können die Stabilität und Konservierung des Weins beeinflussen

"alcohol" = Alkoholgehalt in Prozent

"quality" 

### Was fällt auf? ww.describe()

Ausreißer könnten fixed acidity, residual sugar, free sulfur dioxide und total sulfur dioxide sein; der maximale Wert ist teils ein erhebliches Vielfaches des 75%-Perzentil-Wert

keine fehlenden Werte (bereinigt)

# Aufgabe 2:

## Welche Fragestellungen für Ihren Datensatz können Sie mit der Indizierung und Manipulation                                                                                 bzw. Gruppierungsfunktion lösen?

Wie viel Prozent der Weine haben eine Qualität von 8 oder höher? (8 oder höher entspricht einer sehr hohen Qualität)

-> Nur 180 Weißweine und 18 Rotweine haben eine Qualität von 8 oder höher. Das entspricht 198/6497 Weinproben (~ 3,05% aller getesteten Weine).

In [3]:
# Datensatz Weißwein
ww[ww.quality>=8] # 180 Weißweine haben eine Qualität von 8 oder höher. 

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
17              6.2             0.660         0.48             1.2      0.029   
20              6.2             0.660         0.48             1.2      0.029   
22              6.8             0.260         0.42             1.7      0.049   
68              6.7             0.230         0.31             2.1      0.046   
74              6.7             0.230         0.31             2.1      0.046   
...             ...               ...          ...             ...        ...   
4765            5.2             0.300         0.34             1.5      0.038   
4766            6.4             0.320         0.25             5.0      0.055   
4786            4.4             0.320         0.39             4.3      0.030   
4787            3.9             0.225         0.40             4.2      0.030   
4802            5.8             0.280         0.34             2.2      0.037   

      free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
17                   29.0                  75.0  0.98920  3.33       0.39   
20                   29.0                  75.0  0.98920  3.33       0.39   
22                   41.0                 122.0  0.99300  3.47       0.48   
68                   30.0                  96.0  0.99260  3.33       0.64   
74                   30.0                  96.0  0.99260  3.33       0.64   
...                   ...                   ...      ...   ...        ...   
4765                 18.0                  96.0  0.98942  3.56       0.48   
4766                 28.0                 138.0  0.99171  3.27       0.50   
4786                 31.0                 127.0  0.98904  3.46       0.36   
4787                 29.0                 118.0  0.98900  3.57       0.36   
4802                 24.0                 125.0  0.98986  3.36       0.33   

      alcohol  quality  
17       12.8        8  
20       12.8        8  
22       10.5        8  
68       10.7        8  
74       10.7        8  
...       ...      ...  
4765     13.0        8  
4766     12.4        8  
4786     12.8        8  
4787     12.8        8  
4802     12.8        8  

[180 rows x 12 columns]

In [4]:
# Datensatz Rotwein
rw[rw.quality>=8] # Nur 18 Rotweine haben eine Qualität von 8 oder höher.

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
267             7.9              0.35         0.46             3.6      0.078   
278            10.3              0.32         0.45             6.4      0.073   
390             5.6              0.85         0.05             1.4      0.045   
440            12.6              0.31         0.72             2.2      0.072   
455            11.3              0.62         0.67             5.2      0.086   
481             9.4              0.30         0.56             2.8      0.080   
495            10.7              0.35         0.53             2.6      0.070   
498            10.7              0.35         0.53             2.6      0.070   
588             5.0              0.42         0.24             2.0      0.060   
828             7.8              0.57         0.09             2.3      0.065   
1061            9.1              0.40         0.50             1.8      0.071   
1090           10.0              0.26         0.54             1.9      0.083   
1120            7.9              0.54         0.34             2.5      0.076   
1202            8.6              0.42         0.39             1.8      0.068   
1269            5.5              0.49         0.03             1.8      0.044   
1403            7.2              0.33         0.33             1.7      0.061   
1449            7.2              0.38         0.31             2.0      0.056   
1549            7.4              0.36         0.30             1.8      0.074   

      free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
267                  15.0                  37.0  0.99730  3.35       0.86   
278                   5.0                  13.0  0.99760  3.23       0.82   
390                  12.0                  88.0  0.99240  3.56       0.82   
440                   6.0                  29.0  0.99870  2.88       0.82   
455                   6.0                  19.0  0.99880  3.22       0.69   
481                   6.0                  17.0  0.99640  3.15       0.92   
495                   5.0                  16.0  0.99720  3.15       0.65   
498                   5.0                  16.0  0.99720  3.15       0.65   
588                  19.0                  50.0  0.99170  3.72       0.74   
828                  34.0                  45.0  0.99417  3.46       0.74   
1061                  7.0                  16.0  0.99462  3.21       0.69   
1090                 42.0                  74.0  0.99451  2.98       0.63   
1120                  8.0                  17.0  0.99235  3.20       0.72   
1202                  6.0                  12.0  0.99516  3.35       0.69   
1269                 28.0                  87.0  0.99080  3.50       0.82   
1403                  3.0                  13.0  0.99600  3.23       1.10   
1449                 15.0                  29.0  0.99472  3.23       0.76   
1549                 17.0                  24.0  0.99419  3.24       0.70   

      alcohol  quality  
267      12.8        8  
278      12.6        8  
390      12.9        8  
440       9.8        8  
455      13.4        8  
481      11.7        8  
495      11.0        8  
498      11.0        8  
588      14.0        8  
828      12.7        8  
1061     12.5        8  
1090     11.8        8  
1120     13.1        8  
1202     11.7        8  
1269     14.0        8  
1403     10.0        8  
1449     11.3        8  
1549     11.4        8

Was ist der durchschnittliche Alkoholgehalt für jede Weinqualität?

- genauer in Ausgabe einsehbar (unten)
- tatsächlich haben qualitativ hochwertige Weine auch höheren Alkoholanteil
- geringere Mengen an hochqualitativen Weinen, d. h. Ausreißer haben stärkere Wirkung

In [5]:
# Datensatz Weißwein
ww_mean_alcohol_for_each_quality = ww.groupby('quality')['alcohol'].mean()

print(f"Durchschnittlicher Alkoholgehalt für jede Qualitätsstufe (Stufe 1, 2, 10 sind im Datensatz nicht vergeben):\n"
      f"\n{ww_mean_alcohol_for_each_quality}")

Durchschnittlicher Alkoholgehalt für jede Qualitätsstufe (Stufe 1, 2, 10 sind im Datensatz nicht vergeben):

quality
3    10.345000
4    10.152454
5     9.808840
6    10.575372
7    11.367936
8    11.636000
9    12.180000
Name: alcohol, dtype: float64


In [6]:
# Datensatz Rotwein
rw_mean_alcohol_for_each_quality = rw.groupby('quality')['alcohol'].mean()

print(f"Durchschnittlicher Alkoholgehalt für jede Qualitätsstufe (Stufe 1, 2, 9, 10 sind im Datensatz nicht vergeben):\n"
      f"\n{rw_mean_alcohol_for_each_quality}")

Durchschnittlicher Alkoholgehalt für jede Qualitätsstufe (Stufe 1, 2, 9, 10 sind im Datensatz nicht vergeben):

quality
3     9.955000
4    10.265094
5     9.899706
6    10.629519
7    11.465913
8    12.094444
Name: alcohol, dtype: float64


## Formulieren Sie Hypothesen.

Der Wert für citric acid muss stets geringer sein als der Wert für fixed acidity.

-> trifft zu, Hypothese bestätigt (siehe unten)

-> fixed acidity - citric acid ist stets größer 0

In [7]:
# Datensatz Weißwein
ww_fa_minus_ca = ww['fixed acidity']-ww['citric acid']
# positiv für alle Proben (logisch, da Zitronensäure eine fixe Säure ist)

print(f"Fixe Acidität minus Zitronensäure (Weißwein):\n"
      f"\n{ww_fa_minus_ca}")

Fixe Acidität minus Zitronensäure (Weißwein):

0       6.64
1       5.96
2       7.70
3       6.88
4       6.88
        ... 
4893    5.91
4894    6.24
4895    6.31
4896    5.20
4897    5.62
Length: 4898, dtype: float64


In [8]:
# Datensatz Rotwein
rw_fa_minus_ca = rw['fixed acidity']-rw['citric acid']
# positiv für alle Proben (logisch, da Zitronensäure eine fixe Säure ist)

print(f"Fixe Acidität minus Zitronensäure (Rotwein):\n"
      f"\n{rw_fa_minus_ca}")

Fixe Acidität minus Zitronensäure (Rotwein):

0        7.40
1        7.80
2        7.76
3       10.64
4        7.40
        ...  
1594     6.12
1595     5.80
1596     6.17
1597     5.78
1598     5.53
Length: 1599, dtype: float64


Hohe volatile acidity weißt auf verdorbenen Wein hin, folglich ist die Qualität niedriger.

- trifft nicht eindeutig zu, Differenz zum Durchschnitt eher gering (siehe unten)
- durchschnittliche Qualität bei hoher volatile acidity (über oder gleich dem 75% Perzentil) liegt in beiden Fällen (bei Rot- und Weißwein) ein wenig unter dem Durchschnitt:
- Weißwein: durchschnittl. Qualität = 5.88, durchschnittl. Qualität bei hoher volatile acidity = 5.72
- Rotwein: durchschnittl. Qualität = 5.64, durchschnittl. Qualität bei hoher volatile acidity = 5.27

Niedrige volatile acidity weißt auf fehlendes Aroma hin, folglich ist die Qualität niedriger.

- trifft auch nicht eindeutig zu, Differenz zum Durchschnitt eher gering (siehe unten)
- durchschnittliche Qualität bei niedriger volatile acidity (unter oder gleich dem 25% Perzentil) liegt in beiden Fällen (bei Rot- und Weißwein) ein wenig über dem Durchschnitt:
- Weißwein: durchschnittl. Qualität = 5.88, durchschnittl. Qualität bei niedriger volatile acidity = 6.11
- Rotwein: durchschnittl. Qualität = 5.64, durchschnittl. Qualität bei niedriger volatile acidity = 6.08

In [9]:
# Datensatz Weißwein
mean_weisswein_high_va = ww[ww['volatile acidity']>=0.32]['quality'].mean() 
# gibt Durchschnittsqualität für alle Weißweinproben mit volatile acidity >= 0.32 g/L aus

mean_weisswein_low_va = ww[ww['volatile acidity']<=0.21]['quality'].mean() 
# gibt Durchschnittsqualität für alle Weißweinproben mit volatile acidity <= 0.21 g/L aus

print(f"Durchschnittliche Qualität bei hoher volatiler Acidität (Weißwein): {mean_weisswein_high_va}\n")

print(f"Durchschnittliche Qualität bei niedriger volatiler Acidität (Weißwein): {mean_weisswein_low_va}")

Durchschnittliche Qualität bei hoher volatiler Acidität (Weißwein): 5.723388848660391

Durchschnittliche Qualität bei niedriger volatiler Acidität (Weißwein): 6.109589041095891


In [10]:
# Datensatz Rotwein
mean_rotwein_high_va = rw[rw['volatile acidity']>=0.64]['quality'].mean() 
# gibt Durchschnittsqualität für alle Rotweinproben mit volatile acidity >= 0.64 g/L aus

mean_rotwein_low_va = rw[rw['volatile acidity']<=0.39]['quality'].mean() 
# gibt Durchschnittsqualität für alle Rotweinproben mit volatile acidity <= 0.39 g/L aus

print(f"Durchschnittliche Qualität bei hoher volatiler Acidität (Rotwein): {mean_rotwein_high_va}\n")

print(f"Durchschnittliche Qualität bei niedriger volatiler Acidität (Rotwein): {mean_rotwein_low_va}")

Durchschnittliche Qualität bei hoher volatiler Acidität (Rotwein): 5.2754342431761785

Durchschnittliche Qualität bei niedriger volatiler Acidität (Rotwein): 6.078817733990148


# Aufgabe 3:

## Bestimmen Sie Lage- und Streuungsmaße der Merkmale in Ihrem Datensatz. Verwenden Sie df.describe()

In [11]:
# Datensatz Weißwein
ww.describe()

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    4898.000000       4898.000000  4898.000000     4898.000000   
mean        6.854788          0.278241     0.334192        6.391415   
std         0.843868          0.100795     0.121020        5.072058   
min         3.800000          0.080000     0.000000        0.600000   
25%         6.300000          0.210000     0.270000        1.700000   
50%         6.800000          0.260000     0.320000        5.200000   
75%         7.300000          0.320000     0.390000        9.900000   
max        14.200000          1.100000     1.660000       65.800000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  4898.000000          4898.000000           4898.000000  4898.000000   
mean      0.045772            35.308085            138.360657     0.994027   
std       0.021848            17.007137             42.498065     0.002991   
min       0.009000             2.000000              9.000000     0.987110   
25%       0.036000            23.000000            108.000000     0.991723   
50%       0.043000            34.000000            134.000000     0.993740   
75%       0.050000            46.000000            167.000000     0.996100   
max       0.346000           289.000000            440.000000     1.038980   

                pH    sulphates      alcohol      quality  
count  4898.000000  4898.000000  4898.000000  4898.000000  
mean      3.188267     0.489847    10.514267     5.877909  
std       0.151001     0.114126     1.230621     0.885639  
min       2.720000     0.220000     8.000000     3.000000  
25%       3.090000     0.410000     9.500000     5.000000  
50%       3.180000     0.470000    10.400000     6.000000  
75%       3.280000     0.550000    11.400000     6.000000  
max       3.820000     1.080000    14.200000     9.000000

In [12]:
# Datensatz Rotwein
rw.describe()

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    1599.000000       1599.000000  1599.000000     1599.000000   
mean        8.319637          0.527821     0.270976        2.538806   
std         1.741096          0.179060     0.194801        1.409928   
min         4.600000          0.120000     0.000000        0.900000   
25%         7.100000          0.390000     0.090000        1.900000   
50%         7.900000          0.520000     0.260000        2.200000   
75%         9.200000          0.640000     0.420000        2.600000   
max        15.900000          1.580000     1.000000       15.500000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  1599.000000          1599.000000           1599.000000  1599.000000   
mean      0.087467            15.874922             46.467792     0.996747   
std       0.047065            10.460157             32.895324     0.001887   
min       0.012000             1.000000              6.000000     0.990070   
25%       0.070000             7.000000             22.000000     0.995600   
50%       0.079000            14.000000             38.000000     0.996750   
75%       0.090000            21.000000             62.000000     0.997835   
max       0.611000            72.000000            289.000000     1.003690   

                pH    sulphates      alcohol      quality  
count  1599.000000  1599.000000  1599.000000  1599.000000  
mean      3.311113     0.658149    10.422983     5.636023  
std       0.154386     0.169507     1.065668     0.807569  
min       2.740000     0.330000     8.400000     3.000000  
25%       3.210000     0.550000     9.500000     5.000000  
50%       3.310000     0.620000    10.200000     6.000000  
75%       3.400000     0.730000    11.100000     6.000000  
max       4.010000     2.000000    14.900000     8.000000

(50% = Median, std [standard deviation] = Standartabweichung, mean = Arithmetisches Mittel)

## Haben Sie diese Ergebnisse erwartet?

Im Allgemeinen sind diese Ergebnisse zu erwarten.

## Spannende Erkenntnisse:

- pH ist bei rund 3, was für ein Getränk relativ sauer ist. Ein saurer pH-Wert (kleiner als 7) bei Wein ist allerdings nicht verwunderlich.
- Geringer Effekt des Alkoholgehalts auf die Dichte: Arithmetisches Mittel der Dichte ist rund 1, obwohl durchschnittlich etwa 10% Alkohol im Wein enthalten sind. Die Dichte von Alkohol ist aber bei Raumtemperatur um 20% geringer als die Dichte von Wasser (Ethanol: 0,8 g/cm^3, Wasser: 1 g/cm^3).

## Liegen Mittelwert und Median weit auseinander? Wie groß ist die Streuung?

Median und Mittelwert stehen in fast allen Fällen (bei Weiß- und Rotwein) nah beieinander, was auf eine geringe Streuung hinweist. Ausnahmen sind residual sugar und beide Kategorien des sulfur dioxide: Bei Plan zum späteren Verkauf/Konsum muss sulfur dioxide höher sein, um den Wein lagern zu können. Zucker schwankt aufgrund der Weinsorten stärker, denn ein lieblicher oder halbtrockener Wein enthält mehr Zucker als ein trockener Wein. Leider ist die genauere Weinkategorie nicht im Datensatz angegeben.

# Aufgabe 4:

## Skalenniveaus der Merkmale (Bedeutung von oben ergänzen)

| Merkmal | Skalenniveau | Bedeutung | Einheit |
|---------------|---------------|---------------|---------------|
| fixed acidity | kardinal | stabile Säuren, bestimmen den pH-Wert und Geschmack | g/L |
| volatile acidity | kardinal | in geringen Mengen für Aroma gewünscht, zu viel ist ein Weinfehler | g/L |
| citric acid | kardinal | Teil der fixen Säuren, trägt zur Frische bei | g/L |
| residual sugar | kardinal | unvergorener Zucker im Wein | g/L |
| chlorides | kardinal | Salze, die den Salzgehalt beeinflussen | g/L |
| free sulfur dioxide | kardinal | schützt Wein vor Oxidation und unerwünschten Mikroorganismen | mg/L |
| total sulfur dioxide | kardinal | Summe aus freiem und gebundenem SO2, Gesamtschutz des Weins | mg/L |
| density | kardinal | Dichte des Weins | g/cm^3 |
| pH | kardinal | 1 - 7 heißt sauer, 7 - 14 ist basisch | 1 - 14 |
| sulphates | kardinal | Salze der Schwefelsäure, können die Stabilität und Konservierung des Weins beeinflussen | g/L |
| alcohol | kardinal | Alkoholgehalt | % |
| quality | ordinal / kardinal, wir betrachten es als kardinal | Qualität / Geschmack | 1 - 10 |

# Aufgabe 5:

## Welche Korellationskoeffizienten möchten sie anwenden?
- Bravais-Pearson-Korrelationskoeffizient, da wir hauptsächlich bzw. nur kardinal skalierte Variablen untersuchen

## Prüfen Sie mögliche Zusammenhänge. Waren die Ergebnisse zu erwarten?

In [13]:
kardinal_columns = ['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar', 'chlorides', 'free sulfur dioxide', 
                   'total sulfur dioxide', 'density', 'pH', 'sulphates', 'alcohol', 'quality']

In [14]:
# Datensatz Weißwein
ww[kardinal_columns].corr()

fixed acidity  volatile acidity  citric acid  \
fixed acidity              1.000000         -0.022697     0.289181   
volatile acidity          -0.022697          1.000000    -0.149472   
citric acid                0.289181         -0.149472     1.000000   
residual sugar             0.089021          0.064286     0.094212   
chlorides                  0.023086          0.070512     0.114364   
free sulfur dioxide       -0.049396         -0.097012     0.094077   
total sulfur dioxide       0.091070          0.089261     0.121131   
density                    0.265331          0.027114     0.149503   
pH                        -0.425858         -0.031915    -0.163748   
sulphates                 -0.017143         -0.035728     0.062331   
alcohol                   -0.120881          0.067718    -0.075729   
quality                   -0.113663         -0.194723    -0.009209   

                      residual sugar  chlorides  free sulfur dioxide  \
fixed acidity               0.089021   0.023086            -0.049396   
volatile acidity            0.064286   0.070512            -0.097012   
citric acid                 0.094212   0.114364             0.094077   
residual sugar              1.000000   0.088685             0.299098   
chlorides                   0.088685   1.000000             0.101392   
free sulfur dioxide         0.299098   0.101392             1.000000   
total sulfur dioxide        0.401439   0.198910             0.615501   
density                     0.838966   0.257211             0.294210   
pH                         -0.194133  -0.090439            -0.000618   
sulphates                  -0.026664   0.016763             0.059217   
alcohol                    -0.450631  -0.360189            -0.250104   
quality                    -0.097577  -0.209934             0.008158   

                      total sulfur dioxide   density        pH  sulphates  \
fixed acidity                     0.091070  0.265331 -0.425858  -0.017143   
volatile acidity                  0.089261  0.027114 -0.031915  -0.035728   
citric acid                       0.121131  0.149503 -0.163748   0.062331   
residual sugar                    0.401439  0.838966 -0.194133  -0.026664   
chlorides                         0.198910  0.257211 -0.090439   0.016763   
free sulfur dioxide               0.615501  0.294210 -0.000618   0.059217   
total sulfur dioxide              1.000000  0.529881  0.002321   0.134562   
density                           0.529881  1.000000 -0.093591   0.074493   
pH                                0.002321 -0.093591  1.000000   0.155951   
sulphates                         0.134562  0.074493  0.155951   1.000000   
alcohol                          -0.448892 -0.780138  0.121432  -0.017433   
quality                          -0.174737 -0.307123  0.099427   0.053678   

                       alcohol   quality  
fixed acidity        -0.120881 -0.113663  
volatile acidity      0.067718 -0.194723  
citric acid          -0.075729 -0.009209  
residual sugar       -0.450631 -0.097577  
chlorides            -0.360189 -0.209934  
free sulfur dioxide  -0.250104  0.008158  
total sulfur dioxide -0.448892 -0.174737  
density              -0.780138 -0.307123  
pH                    0.121432  0.099427  
sulphates            -0.017433  0.053678  
alcohol               1.000000  0.435575  
quality               0.435575  1.000000

### erwähnenswerte Zusammenhänge im Weißwein - Datensatz:
- fixed acidity - citric acid (0.29, denn citric acid ist ein Teil der fixed acidity)
- fixed acidity - pH (-0.43, denn Menge an Säure im Wein bestimmt pH-Wert)
- residual sugar und free sulfur dioxide (0.30)
- residual sugar - total sulfur dioxide (0.40)
- residual sugar - density (0.84, denn gelöster Zucker beinflusst Dichte)
- residual sugar - alcohol (-0.45, denn Zucker gärt zu Alkohol)
- chlorides - alcohol (-0.36)
- free sulfur dioxide - total sulfur dioxide (0.62, denn free sulfur dioxide ist ein Teil der total sulfur dioxide)
- total sulfur dioxide - density(0.53)
- total sulfur dioxide - alcohol (-0,45)
- density - alcohol (-0,78)
- density - quality (-0.31)
- alcohol - quality (0.44)

### erwähnenswerte Zusammenhänge im Rotwein - Datensatz:
- fixed acidity - citric acid (0.67, denn citric acid ist ein Teil der fixed acidity)
- fixed acidity - density (0.67)
- fixed acidity - pH (-0.68, denn Menge an Säure im Wein bestimmt pH-Wert)
- volatile acidity - citric acid (-0.55)
- volatile acidity - quality (-0.39)
- citric acid - density (0.36)
- citric acid - pH (-0.54, , denn Menge an Säure im Wein bestimmt pH-Wert)
- residual sugar - density (0.36, denn gelöster Zucker beinflusst Dichte)
- chlorides - sulphates (0.37)
- free sulfur dioxide - total sulfur dioxide (0.67, denn free sulfur dioxide ist ein Teil der total sulfur dioxide)
- density - alcohol (-0.50)
- alcohol - quality (0.48)

Zusammenhänge von z. B. fixed acidity und citric acid (beide Datensätze) waren abzusehen, da citric acid ein Bestandteil der fixed acidity ist. Hingegen andere Korrelationen wie z. B. zwischen residual sugar und sulfur dioxide (Weißwein - Datensatz) sind neue Ergebnisse. Es gibt also sowohl klar absehbare als auch neue Korrelationen.

In [21]:
# Datensatz Rotwein
rw[kardinal_columns].corr()

fixed acidity  volatile acidity  citric acid  \
fixed acidity              1.000000         -0.256131     0.671703   
volatile acidity          -0.256131          1.000000    -0.552496   
citric acid                0.671703         -0.552496     1.000000   
residual sugar             0.114777          0.001918     0.143577   
chlorides                  0.093705          0.061298     0.203823   
free sulfur dioxide       -0.153794         -0.010504    -0.060978   
total sulfur dioxide      -0.113181          0.076470     0.035533   
density                    0.668047          0.022026     0.364947   
pH                        -0.682978          0.234937    -0.541904   
sulphates                  0.183006         -0.260987     0.312770   
alcohol                   -0.061668         -0.202288     0.109903   
quality                    0.124052         -0.390558     0.226373   

                      residual sugar  chlorides  free sulfur dioxide  \
fixed acidity               0.114777   0.093705            -0.153794   
volatile acidity            0.001918   0.061298            -0.010504   
citric acid                 0.143577   0.203823            -0.060978   
residual sugar              1.000000   0.055610             0.187049   
chlorides                   0.055610   1.000000             0.005562   
free sulfur dioxide         0.187049   0.005562             1.000000   
total sulfur dioxide        0.203028   0.047400             0.667666   
density                     0.355283   0.200632            -0.021946   
pH                         -0.085652  -0.265026             0.070377   
sulphates                   0.005527   0.371260             0.051658   
alcohol                     0.042075  -0.221141            -0.069408   
quality                     0.013732  -0.128907            -0.050656   

                      total sulfur dioxide   density        pH  sulphates  \
fixed acidity                    -0.113181  0.668047 -0.682978   0.183006   
volatile acidity                  0.076470  0.022026  0.234937  -0.260987   
citric acid                       0.035533  0.364947 -0.541904   0.312770   
residual sugar                    0.203028  0.355283 -0.085652   0.005527   
chlorides                         0.047400  0.200632 -0.265026   0.371260   
free sulfur dioxide               0.667666 -0.021946  0.070377   0.051658   
total sulfur dioxide              1.000000  0.071269 -0.066495   0.042947   
density                           0.071269  1.000000 -0.341699   0.148506   
pH                               -0.066495 -0.341699  1.000000  -0.196648   
sulphates                         0.042947  0.148506 -0.196648   1.000000   
alcohol                          -0.205654 -0.496180  0.205633   0.093595   
quality                          -0.185100 -0.174919 -0.057731   0.251397   

                       alcohol   quality  
fixed acidity        -0.061668  0.124052  
volatile acidity     -0.202288 -0.390558  
citric acid           0.109903  0.226373  
residual sugar        0.042075  0.013732  
chlorides            -0.221141 -0.128907  
free sulfur dioxide  -0.069408 -0.050656  
total sulfur dioxide -0.205654 -0.185100  
density              -0.496180 -0.174919  
pH                    0.205633 -0.057731  
sulphates             0.093595  0.251397  
alcohol               1.000000  0.476166  
quality               0.476166  1.000000